3-Basic Elements Update

EricoDeMecha opened the pull request:
This PR does the following 
 - Updates sub-lectures CMake 
 - Updates Qt5 resource to cmake
 - Updates and formats QML files.

rutura reviewed (changes requested):
The use of Columns, Models and Repeaters can be confusing to people starting out. It can steal focus from the main point of the lecture: Image. 

Needed changes: 

- Keep the code simple in this lecture (as it was)
- Add another lecture with your changes  (put is last), using The column element and repeater just like you did here. That way, the learner is incrementally exposed to new practical concepts.

rutura reviewed (changes requested):
This lecture needs to be extended with other concepts that should go in added lectures: 

- Achieve the same using the inline QML built in Component: https://doc.qt.io/qt-6/qml-qtqml-component.html
- How does this component concept contrast with the QML modules concept. May be add a new lecture implementing the same functionality, but the button is brought in as a QML module?

rutura reviewed (changes requested):
The Image element is the first opportunity to introduce external resources into the project. Here is an idea. We set up two separate projects: 

- One loads the image from the old Qt resource system
- The other loads them through the qt_add_resources cmake command.
This is just showing the two methods to the learner. In the remainder of the project, we'll stick to qt_add_resources.

EricoDeMecha commented:
- I have restored the old resource system in lecture 2-Image and introduced another lecture on the new cmake resource system on lecture 21-Image-cmake-resource-system.
- Also removed repeaters and columns in this lecture.

EricoDeMecha commented:
- Added a couple of lectures building up incrementally
-  Introduced inlined components on lecture 3-CustomComponents(now lecture 7-Customcomponents)

rutura commented:
Please ONLY submit the pull request when all the suggested [points ](https://github.com/rutura/Qt6QMLBeginnersCodePrep/pull/3#issuecomment-1892629668)have been addressed.

EricoDeMecha commented:
Added 3 more lectures on QML modules.

rutura commented on 3.BasicElements/8-QMLModules/CMakeLists.txt:
```diff
@@ -0,0 +1,54 @@
+        Main.qml
+)
+
+add_subdirectory(components)
```
Documentation for how the lectures and changes you add is needed together with relevant links is needed. 
Example: 

- How are entities in the component imported in the Main.qml file. How to set the name that one imports.
- Why is the qmldir file needed, how does the information in there affect the project? 
- How does this way of doing things compare to other techniques like Components? 

Try to think from the perspective of someone learning these things for the first time and answer the questions they might have. I expect detailed documentation like this for all changes that you push.

rutura commented on 3.BasicElements/9-QMLModulesProperties/CMakeLists.txt:
```diff
@@ -0,0 +1,55 @@
+)
+
+add_subdirectory(components)
+add_subdirectory(constants)
```
Documentation again. Make it extensive and compare with other techniques introduced in the chapter. Provide links to the reference documentation. Please remember this in subsequent PRs that you send. Only send the PR when the changes are properly documented.

rutura commented on 3.BasicElements/10-QMLModulesUtilities/CMakeLists.txt:
```diff
@@ -0,0 +1,56 @@
+
+add_subdirectory(components)
+add_subdirectory(constants)
+add_subdirectory(utilities)
```
Documentation!!

EricoDeMecha replied:
Updated, check the new changes.

rutura reviewed (changes requested):
Overall, the changes you've submitted are nice. The last step is to properly document the code: 

- Why one should use the technique
- How do changes in code affect the project
- Think of it as an imaginary flowchart following the application from the entry point up to the concept that we're trying to explain.
- Provide links to relevant parts in the official docs, learners expect this.
- Also, compare the techniques,

I expect this to be our standard going forward.

Please document all the changes you did in terms of Components and QML modules. Then, update the PR.
